File: ctci/tmp/chapter_5_bit_manipulation/cpp/5.5-debugger.cpp
#include <iostream>

using namespace std;

/**
 * Determines if a number is a power of two.
 */
bool is_power_of_two(unsigned int n)
{
    return n && !(n & (n - 1));
}

int main()
{
    cout << is_power_of_two(0) << endl; // 0
    cout << is_power_of_two(2) << endl; // 1
    cout << is_power_of_two(6) << endl; // 0
    cout << is_power_of_two(8) << endl; // 1

    return 0;
}

File: ctci/tmp/chapter_5_bit_manipulation/js/5.5-debugger.js
/**
 * Determines if a number is a power of two.
 * @param   {number}  n
 * @returns {boolean}
 */
const isPowerOfTwo = n => {
  return n && !(n & (n - 1))
}

console.log(isPowerOfTwo(0)) // false
console.log(isPowerOfTwo(2)) // true
console.log(isPowerOfTwo(6)) // false
console.log(isPowerOfTwo(8)) // true
